apps/voip/3cx/restapi/custom/api.pm

```diff
@@ -194,6 +194,12 @@ sub request_api {
         warning_status => '', unknown_status => '', critical_status => '%{http_code} < 200 or %{http_code} >= 300'
     );
 
+    # Some content may be strangely returned, for example : "[{\"Category\":\"provider\",\"Count\":1}]"
+    if ($content =~ /^"(\[.*\])"$/) {
+        $content = $1;
+        $content =~ s/\\"/"/g;
+    }
```

Comment:
I can't accept that. It can break json

Reply:
The added API call returns `"[{\"Category\":\"provider\",\"Count\":1}]"`.
This is not JSON compliant, this is why we have to remove leading and trailing `"`, and remove escaping `\` characters before `"`, but of course only if `$content` matches this strange content format.

Reply:
The JSON is not compliant ? So that's a bug of 3cx ?

Reply:
I think I will not manage to get 3CX to correct this, as it is not documented nor supported :-|
In addition, this is not really a bug, as the JSON string is simply surrounded by `"`, which could in some kind be "expected". We simply have to remove the surrounding...

Reply:
Could you provide me the raw string result of that API call ?

Reply:
Here it is :
`"[{\"Category\":\"provider\",\"Count\":1}]"`

Reply:
If no update is available, I get :
`"[]"`

Reply:
The string seems ok. You should can decode with json without problem (no change is needed)

Reply:
You do the print with Data::Dumper ?

Reply:
Without this trick the exception is triggered :
`Cannot decode json response: JSON text must be an object or array (but found number, string, true, false or null, use allow_nonref to allow this)`

Reply:
I print response content using `--debug`.
This is also what curl returns.

Reply:
With Data::Dumper :
`$VAR1 = '"[{\\"Category\\":\\"provider\\",\\"Count\\":1}]"';`